framework/sysdeps/windows/tmpfile.c

```diff
@@ -43,19 +43,30 @@ static unsigned next_random()
 int open_memfd(enum MemfdCloexecFlag flag)
 {
     wchar_t tmpname[sizeof SANDSTONE_STRINGIFY(UINT_MAX) ".tmp"];
+    wchar_t tmppath[MAX_PATH];
+
     HANDLE hFile = INVALID_HANDLE_VALUE;
     SECURITY_ATTRIBUTES sa = {};
     sa.nLength = sizeof(sa);
     sa.bInheritHandle = (flag == MemfdInheritOnExec);
     sa.lpSecurityDescriptor = NULL;
 
+    DWORD getTempPathRes = GetTempPathW(MAX_PATH, tmppath);
+
+    if (getTempPathRes == 0)
+        return -1;
+
     for (int i = 0; hFile == INVALID_HANDLE_VALUE && i < 16; ++i) {
         wcscat(_ultow(next_random(), tmpname, 36), L".tmp");            // yes, base 36
         DWORD access = GENERIC_READ | GENERIC_WRITE;
         DWORD sharemode = FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE;
         DWORD creation = CREATE_NEW;
         DWORD flags = FILE_ATTRIBUTE_TEMPORARY | FILE_FLAG_DELETE_ON_CLOSE;
-        hFile = CreateFileW(tmpname, access, sharemode, &sa, creation, flags, NULL);
+
+        if ((wcslen(tmppath) + wcslen(tmpname)) > MAX_PATH)
```

Comment:
Shouldn't it be `>= MAX_PATH`?

Reply:
If I count the sizes right we have a buffer (tmppath) sized MAX_PATH. If we put something with MAX_PATH length string we shouldn't have a buffer overflow.

Reply:
But we need one more extra for the null terminator, don't we?

Reply:
`CreateFileW` limits the filename to `MAX_PATH`, but spec is vague on details if this includes `<NUL>` or not. `wcslen` handles non-null terminated strings, so `(wcslen(tmppath) + wcslen(tmpname)) == MAX_PATH` case should technically be fine, but I'd feel better about this if we kept the one extra slot for null termination.

Reply:
Also, from https://en.cppreference.com/w/cpp/string/wide/wcscat:
```
The behavior is undefined if the destination array is not large enough for the contents of
both str and dest and the terminating null wide character.
```

Reply:
Done